File: sms/quickstart/quickstart.py
"""
SMS Quickstart — send your first text in 60 seconds.

Steps:
  1. List phone numbers to find one with SMS capability
  2. Send a test SMS to TEST_PHONE_NUMBER
  3. Print the delivery receipt

Usage:
  python quickstart.py
"""

import os
import sys
from dotenv import load_dotenv
from commune import CommuneClient

load_dotenv()

# Validate required environment variables before doing anything else.
# Failing fast here gives a clear error rather than a confusing API response.
required = ["COMMUNE_API_KEY", "TEST_PHONE_NUMBER"]
for var in required:
    if not os.environ.get(var):
        raise SystemExit(f"Missing env var: {var} — copy .env.example to .env and fill it in.")

API_KEY = os.environ["COMMUNE_API_KEY"]
TEST_PHONE = os.environ["TEST_PHONE_NUMBER"]

commune = CommuneClient(api_key=API_KEY)


def find_sms_capable_number():
    """Return the first phone number on the account that has SMS capability."""
    numbers = commune.phone_numbers.list()

    if not numbers:
        raise SystemExit(
            "No phone numbers found on your account. "
            "Provision one at https://commune.sh before running this script."
        )

    for number in numbers:
        # Skip numbers that don't have SMS enabled (e.g. voice-only lines)
        if number.capabilities and number.capabilities.sms:
            return number

    raise SystemExit(
        "No SMS-capable phone numbers found. "
        "Check your Commune dashboard and ensure at least one number has SMS enabled."
    )


def main() -> None:
    print("Fetching phone numbers...")
    phone = find_sms_capable_number()
    print(f"Using phone number: {phone.number}  (id={phone.id})")

    print(f"\nSending SMS to {TEST_PHONE}...")
    try:
        result = commune.sms.send(
            to=TEST_PHONE,
            body="Hello from Commune! Your SMS quickstart is working.",
            phone_number_id=phone.id,
        )
    except Exception as e:
        raise SystemExit(f"SMS send failed: {e}")

    # Print the full delivery receipt so you can verify everything looks right.
    print("\nDelivery receipt:")
    print(f"  message_id      = {result.message_id}")
    print(f"  thread_id       = {result.thread_id}")
    print(f"  status          = {result.status}")
    print(f"  credits_charged = {result.credits_charged}")
    print(f"  segments        = {result.segments}")
    print("\nDone. Check your phone for the message.")


if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        print("\nCancelled.")
        sys.exit(0)

File: sms/quickstart/quickstart.ts
/**
 * SMS Quickstart — send your first text in 60 seconds (TypeScript).
 *
 * Steps:
 *   1. List phone numbers to find one with SMS capability
 *   2. Send a test SMS to TEST_PHONE_NUMBER
 *   3. Print the delivery receipt
 *
 * Usage:
 *   npx tsx quickstart.ts
 */

import { CommuneClient } from "commune-ai";
import * as dotenv from "dotenv";

dotenv.config();

// Validate required environment variables before doing anything else.
const required = ["COMMUNE_API_KEY", "TEST_PHONE_NUMBER"];
for (const varName of required) {
  if (!process.env[varName]) {
    console.error(
      `Missing env var: ${varName} — copy .env.example to .env and fill it in.`
    );
    process.exit(1);
  }
}

const API_KEY = process.env.COMMUNE_API_KEY!;
const TEST_PHONE = process.env.TEST_PHONE_NUMBER!;

const commune = new CommuneClient({ apiKey: API_KEY });

async function findSmsCapableNumber() {
  const numbers = await commune.phoneNumbers.list();

  if (!numbers || numbers.length === 0) {
    console.error(
      "No phone numbers found on your account. " +
        "Provision one at https://commune.sh before running this script."
    );
    process.exit(1);
  }

  // Use the first number that has SMS enabled
  for (const number of numbers) {
    if (number.capabilities?.sms) {
      return number;
    }
  }

  console.error(
    "No SMS-capable phone numbers found. " +
      "Check your Commune dashboard and ensure at least one number has SMS enabled."
  );
  process.exit(1);
}

async function main(): Promise<void> {
  console.log("Fetching phone numbers...");
  const phone = await findSmsCapableNumber();
  console.log(`Using phone number: ${phone.number}  (id=${phone.id})`);

  console.log(`\nSending SMS to ${TEST_PHONE}...`);

  let result;
  try {
    result = await commune.sms.send({
      to: TEST_PHONE,
      body: "Hello from Commune! Your SMS quickstart is working.",
      phoneNumberId: phone.id,
    });
  } catch (err) {
    console.error("SMS send failed:", err);
    process.exit(1);
  }

  // Print the full delivery receipt so you can verify everything looks right.
  console.log("\nDelivery receipt:");
  console.log(`  message_id      = ${result.messageId}`);
  console.log(`  thread_id       = ${result.threadId}`);
  console.log(`  status          = ${result.status}`);
  console.log(`  credits_charged = ${result.creditsCharged}`);
  console.log(`  segments        = ${result.segments}`);
  console.log("\nDone. Check your phone for the message.");
}

main().catch((err) => {
  console.error("Unexpected error:", err);
  process.exit(1);
});
